runtime/metrics_api.go

```diff
@@ -172,6 +172,10 @@ func (m *MetricsApi) Timer(name string) Timer {
 	return m.reg.GetOrRegister(name, metrics.NewTimer()).(metrics.Timer)
 }
 
+func (m *MetricsApi) ResettingTimer(name string) ResettingTimer {
+	return m.reg.GetOrRegister(name, NewResettingTimer()).(ResettingTimer)
```

Comment:
@raulk I am not sure how we are going to extend the go-metrics library to support new types, because:

```
func (r *StandardRegistry) register(name string, i interface{}) error {
	if _, ok := r.metrics[name]; ok {
		return DuplicateMetric(name)
	}
	switch i.(type) {
	case Counter, Gauge, GaugeFloat64, Healthcheck, Histogram, Meter, Timer:
		r.metrics[name] = i
	}
	return nil
}
```

Reply:
@nonsense 

`Counter, Gauge, GaugeFloat64, Healthcheck, Histogram, Meter, Timer` <= these are interfaces, right? As long our new structs implement those interfaces, we should be good to go?

Reply:
As discussed, renaming `ResettingTimer` to `ResettingHistogram` and making sure it implements the `Histogram` interface.